src/packet.rs

```diff
@@ -429,6 +433,35 @@ impl Blob {
     }
 }
 
+#[cfg(test)]
+pub fn make_consecutive_blobs(
```

Comment:
This gnarly untested function still bugs me. It looks independently useful, and too similar to `Block::to_blobs()`:

https://github.com/solana-labs/solana/blob/407b2682e85f8fed9584607eda9a0203c84e00db/src/ledger.rs#L644

Looks like `Block::to_blobs()` ought to `enumerate()` as you did and set `Some(blob_index)`. And if so, maybe that `Some` wrapper is worthless and should be removed?  Likewise, wondering why `me_id` and `addr` need to be wrapped in `Option` instead of using `Pubkey::default()` and a default `SocketAddr`. I'm guessing there needs to be a `Block::to_blobs_with_id(recycler, id, addr)` that's used by both the existing `Block::to_blobs()` and this function?

I'm not certain if changes are really needed here, but can you take a closer look at `Block::to_blobs()` and at the very least, add a code comment that acknowledges its existence and explains why it can't be used.